### DISCUSSION TITLE
MacOS USB device passthrough to a Docker container

### DISCUSSION THREAD

**Discussion by skandragon** [rancher-sandbox/rancher-desktop#3904] [Q&A] [answered]
I'm attempting to pass through a USB device from my Mac to a Docker container, but it seems to not be working.  I suspect that is because there is a VM running the actual Docker magic, but perhaps I misunderstand the architecture.

I tried using a file (volume) mapping but the USB device doesn't seem to appear in my Mac's `/dev`, so I'd have to export the specific device to qemu.  Is this possible?

**Comment by jandubois** [comment, marked as answer]
Yes, the containers run inside a VM, and there is no way[^1] to pass through a USB device to the VM.

It is possible to mount additional filesystems into the VM, but that won't give you access to the device itself.

[^1]: Well, https://github.com/lima-vm/lima/pull/1317 kind of shows how it can be made to work, but it requires running various pieces as `root`, which would not be acceptable for Rancher Desktop.

**Reply by skandragon** [reply]
I was afraid of that, thanks.  I guess I need to try this rpi 3b+ sitting here, gaining value daily.  :)
